exercises/practice/killer-sudoku-helper/KillerSudokuHelper.tests.ps1

```diff
@@ -0,0 +1,99 @@
+BeforeAll {
+    . "./KillerSudokuHelper.ps1"
+}
+
+Describe "KillerSudokuHelper test cases" {
+    Context "Trivial 1-digit cages" {
+        It "Trivial 1-digit cages -> 1" {
+            $got  = Invoke-KillerSudokuHelper -Sum 1 -Size 1 -Exclude @()
+            $want = @( ,@(1))
+    
+            ,$got | Should -BeExactly $want
+        }
+    
+        It "Trivial 1-digit cages -> 2" {
+            $got  = Invoke-KillerSudokuHelper -Sum 2 -Size 1 -Exclude @()
+            $want = @( ,@(2))
+    
+            ,$got | Should -BeExactly $want
+        }
+    
+        It "Trivial 1-digit cages -> 3" {
+            $got  = Invoke-KillerSudokuHelper -Sum 3 -Size 1 -Exclude @()
+            $want = @( ,@(3))
+    
+            ,$got | Should -BeExactly $want
+        }
+    
+        It "Trivial 1-digit cages -> 4" {
+            $got  = Invoke-KillerSudokuHelper -Sum 4 -Size 1 -Exclude @()
+            $want = @( ,@(4))
+    
+            ,$got | Should -BeExactly $want
+        }
+    
+        It "Trivial 1-digit cages -> 5" {
+            $got  = Invoke-KillerSudokuHelper -Sum 5 -Size 1 -Exclude @()
+            $want = @( ,@(5))
+    
+            ,$got | Should -BeExactly $want
+        }
+    
+        It "Trivial 1-digit cages -> 6" {
+            $got  = Invoke-KillerSudokuHelper -Sum 6 -Size 1 -Exclude @()
+            $want = @( ,@(6))
+    
+            ,$got | Should -BeExactly $want
+        }
+    
+        It "Trivial 1-digit cages -> 7" {
+            $got  = Invoke-KillerSudokuHelper -Sum 7 -Size 1 -Exclude @()
+            $want = @( ,@(7))
+    
+            ,$got | Should -BeExactly $want
+        }
+    
+        It "Trivial 1-digit cages -> 8" {
+            $got  = Invoke-KillerSudokuHelper -Sum 8 -Size 1 -Exclude @()
+            $want = @( ,@(8))
+    
+            ,$got | Should -BeExactly $want
+        }
+    
+        It "Trivial 1-digit cages -> 9" {
+            $got  = Invoke-KillerSudokuHelper -Sum 9 -Size 1 -Exclude @()
+            $want = @( ,@(9))
+    
+            ,$got | Should -BeExactly $want
+        }
+    }
+
+    It "Cage with sum 45 contains all digits 1:9" {
+        $got  = Invoke-KillerSudokuHelper -Sum 45 -Size 9 -Exclude @()
+        $want = @( ,@(1, 2, 3, 4, 5, 6, 7, 8, 9))
+
+        ,$got | Should -BeExactly $want
+    }
+
+    It "Cage with only 1 possible combination" {
+        $got  = Invoke-KillerSudokuHelper -Sum 7 -Size 3 -Exclude @()
+        $want = @( ,@(1, 2, 4))
```

Comment:
I think it is sorta ugly having the `,` before the array, when I removed the `,` mentioned in the comment above, it does work to remove that.

Reply:
Removing the "," indeed the test still work. My reason for it was how powershell unroll their array depending on context.
 $want = @( @(1, 2, 4))
 $want.Count  will be equal 3.
However  $want = @( ,@(1, 2, 4))
 $want.Count  will be equal 1.

 Technically speaking, want count should only be 1, because there is only one possible way to fill out the cells.
 
 For this problem, it probably doesnt matter regardless when the array solution only consist of one solution of inner array.
 But I think there should be an exercise to teach students about this behavior later